crates/ide-completion/src/context.rs

```diff
@@ -1285,8 +1306,9 @@ impl<'a> CompletionContext<'a> {
                 .map(|it| it.parent_path());
             path_ctx.qualifier = path.map(|path| {
                 let res = sema.resolve_path(&path);
-                let is_super_chain = iter::successors(Some(path.clone()), |p| p.qualifier())
-                    .all(|p| p.segment().and_then(|s| s.super_token()).is_some());
+                let super_chain_len = iter::successors(Some(path.clone()), |p| p.qualifier())
+                    .filter_map(|p| p.segment().and_then(|s| s.super_token()))
+                    .count();
```

Comment:
So we only want to complete `super` when the entire path consists of `super` segments, your changes here just don't count non-super path which isn't quite correct, I think `super_chain_len` needs to be `Option<usize>`, None for when the path consists of something other than super segments, `Some` if its only super segments. An empty path would be `Some(0)`

Reply:
That's a nice catch, thanks for pointing it out!

Now resolved